src/Avalonia.Base/Rendering/ImmediateRenderer.cs

```diff
@@ -115,6 +126,11 @@ public static void Render(DrawingContext context, Visual visual, Rect clipRect)
                             Render(context, child, childClipRect);
                         }
                     }
+
+                    if(platformContext != null)
+                    {
+                        platformContext.RenderOptions = currentRenderOptions;
+                    }
```

Comment:
nit: an exception in ImmediateRenderer code won't crash the app depending on the usage, so it probably makes sense to restore context properties in finally block to make it safe